sdk/src/utils.ts

```diff
@@ -226,6 +228,23 @@ export const getDeploymentData = async (
   }
 }
 
+export const calculatePriceImpact = (
+  startingSqrtPrice: SqrtPrice,
+  endingSqrtPrice: SqrtPrice
+): Percentage => {
+  const startingPrice = startingSqrtPrice.v * startingSqrtPrice.v
+  const endingPrice = endingSqrtPrice.v * endingSqrtPrice.v
+
+  const nominator = BigInt(Math.abs(Number(startingPrice - endingPrice)))
+  const denominator = BigInt(Math.max(Number(startingPrice), Number(endingPrice)))
+
+  return {
+    v: BigInt(
+      Math.round((Number(nominator) / Number(denominator)) * Number(getPercentageDenominator()))
```

Comment:
Check if domain is safe. If not, use native `bigint`. Math.round you can easily replace will `mul_up`